.github/workflows/generate-tidal-api.yml

```diff
@@ -0,0 +1,123 @@
+name: Generate API in the TidalAPI module and create a PR if there are changes
+
+on:
+  workflow_dispatch: # manual triggering of this workflow
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  codegen:
+    runs-on: macos-latest
+    env:
+      API_FOLDER: Sources/TidalAPI/
+      GENERATED_FOLDER_NAME: Generated
+      INPUT_FOLDER_NAME: Config/input
+      SCRIPT_NAME: generate_and_clean.sh
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Step 2: Install OpenAPI Generator via Homebrew
+      - name: Install OpenAPI Generator with Homebrew
+        run: |
+          brew update
+          brew install openapi-generator
+
+      - name: Run Code Generation Script
+        run: |
+          cd $API_FOLDER
+          chmod +x $SCRIPT_NAME
+          ./$SCRIPT_NAME
```

Comment:
Can you also run `./$API_FOLDER/$SCRIPT_NAME`?
If not, the script might not be robust enough.

Reply:
It is not working 🙂 

The script implies that we are in the API folder now, and all the paths in it are relative to the folder.

If you have suggestions, I would be happy to improve it (in follow-up PRs) 😉 

![Screenshot 2024-10-29 at 11 20 39](https://github.com/user-attachments/assets/0ea4b34b-865a-464f-b7f7-7bdaa3da35ea)

Reply:
yes, you can take a look at the script on the Android side. I had the same issues, and changed it so in the beginning it evaluates the directories we need and stores them in variables, that then can be referenced.